packages/protocol/contracts/governance/VestingInstance.sol

```diff
@@ -0,0 +1,379 @@
+   */
+  function withdraw() external nonReentrant onlyBeneficiary {
+    bool isUnpaused = !paused || (paused && block.timestamp >= pauseEndTime);
+    require(isUnpaused, "Withdrawals only allowed in the unpaused state");
+    if (isUnpaused) paused = false;
+    uint256 withdrawableAmount = getWithdrawableAmountAtTimestamp(block.timestamp);
+    require(withdrawableAmount > 0, "No gold is due for withdrawal");
+    currentlyWithdrawn = currentlyWithdrawn.add(withdrawableAmount);
+    require(getGoldToken().transfer(beneficiary, withdrawableAmount), "Withdrawal of gold failed");
+  }
+
+  /**
+   * @notice Allows only the revoker to revoke the vesting. Gold already vested
+   * remains in the contract, the rest is returned to the _refundDestination.
+   * @param revokeTimestamp the revocation timestamp
+   * @dev If revokeTimestamp is less than the current block timestamp, it is set equal to the latter
+   */
+  function revoke(uint256 revokeTimestamp) external nonReentrant onlyRevoker {
+    require(revocable, "Revoking is not allowed");
+    require(!revoked, "Vesting already revoked");
+    revokeTime = revokeTimestamp > block.timestamp ? revokeTimestamp : block.timestamp;
+    uint256 balance = getGoldToken().balanceOf(address(this));
+    uint256 withdrawableAmount = getWithdrawableAmountAtTimestamp(revokeTime);
+    uint256 refundAmount = balance.sub(withdrawableAmount);
+    revoked = true;
+    require(
+      getGoldToken().transfer(refundDestination, refundAmount),
+      "Transfer of refund upon revokation failed"
+    );
+    emit VestingRevoked(revokeTime);
+  }
+
+  /**
+   * @notice Allows only the revoker to pause the gold withdrawal
+   * @param pausePeriod the period for which the withdrawal shall be paused
+   */
+  function pause(uint256 pausePeriod) external onlyRevoker {
+    require(!paused, "Vesting withdrawals already paused");
+    require(revocable, "Vesting must be revokable");
+    require(!revoked, "Vesting already revoked");
+    require(pausePeriod <= 365 days, "Pause period is limited to max. 365 days");
+    paused = true;
+    pauseEndTime = block.timestamp.add(pausePeriod);
+    emit WithdrawalPaused(pausePeriod, block.timestamp);
+  }
+
+  /**
+   * @dev Calculates the amount that has already vested but hasn't been withdrawn yet.
+   * @param timestamp the timestamp at which the calculate the withdrawable amount
+   * @return The withdrawable amount at the timestamp
+   * @dev Function is also made public in order to be called for informational purpose
+   */
+  function getWithdrawableAmountAtTimestamp(uint256 timestamp) public view returns (uint256) {
+    return calculateFreeAmountAtTimestamp(timestamp).sub(currentlyWithdrawn);
+  }
+
+  /**
+   * @dev Calculates the amount that has already vested.
+   * @param timestamp the timestamp at which the calculate the already vested amount
+   */
+  function calculateFreeAmountAtTimestamp(uint256 timestamp) private view returns (uint256) {
```

Comment:
Naming nit: I would go with something like "VestedAmount" to keep the same naming as elsewhere.